app/models/conditions_response/backup.rb

```diff
@@ -274,4 +285,78 @@ def self.create_files_backup_maker(config)
 
     files_backup_maker
   end
+
+
+  # Iterate through the list, rescuing errors. If it's a Slack error,
+  # log and raise the error (stop iterating).
+  # For any other error, log  and send a notification
+  # and continue iterating via the 'next' keyword.
+  #
+  # @param [Enumerable] list - items we are iterating through
+  # @param [String] additional_error_info - additional information to include
+  #                  when logging or notifying
+  # @param [Log] log - the log to write to
+  #
+  def self.iterate_and_log_notify_errors(list, additional_error_info, log)
+
+    list.each do |item|
+      yield(item)
+
+    rescue Slack::Notifier::APIError => slack_error
+      # Halt the backup if we cannot write to Slack; log then raise the error
+      log_slack_error(slack_error, log, "#{additional_error_info}. Current item: #{item.inspect}")
+      raise slack_error
+
+    rescue => backup_error
+      log_and_notify(backup_error, log, "#{additional_error_info}. Current item: #{item.inspect}")
+      next
+    end
+  end
+
+
+  # Record the error and additional_info to the given log and send a Slack notification.
+  #
+  # TODO  this seems like a general-purpose method that we need in many places.  Refactor into a class/module to be available all places
+  #
+  #
+  # @param [Error] error - Error that needs to be recorded
+  # @param [Log] log - the log to write to. Must respond to :error(message)
+  # @param [String] additional_info - any additional information that should also be recorded. Default = ''
+  #
+  def self.log_and_notify(error, log, additional_info = '')
+
+    log_string = additional_info&.empty? ? error.to_s : "#{error} #{additional_info}"
```

Comment:
`additional_info.blank?` would negate the need for the safe navigation operator.

Reply:
yes, of course.  (I'd been working on something outside of Rails and had "don't use Rails specific methods" in my head.)   
Will fix.